spynnaker/pyNN/models/neuron/population_machine_common.py

```diff
@@ -0,0 +1,193 @@
+# Copyright (c) 2017-2020The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from collections import namedtuple
+from spinn_utilities.overrides import overrides
+from spinn_utilities.abstract_base import abstractmethod
+from pacman.model.graphs.machine import MachineVertex
+
+from spinn_front_end_common.interface.provenance import (
+    ProvidesProvenanceDataFromMachineImpl)
+from spinn_front_end_common.interface.buffer_management.buffer_models import (
+    AbstractReceiveBuffersToHost)
+from spinn_front_end_common.utilities.helpful_functions import (
+    locate_memory_region_for_placement)
+from spinn_front_end_common.interface.profiling.profile_utils import (
+    get_profiling_data, reserve_profile_region, write_profile_region_data)
+from spinn_front_end_common.abstract_models import (
+    AbstractRecordable, AbstractHasAssociatedBinary)
+from spinn_front_end_common.interface.buffer_management\
+    .recording_utilities import (
+        get_recording_header_size, get_recording_header_array)
+from spinn_front_end_common.interface.simulation.simulation_utilities import (
+    get_simulation_header_array)
+
+from spinn_front_end_common.utilities.utility_objs import ExecutableType
+from spinn_front_end_common.interface.profiling import AbstractHasProfileData
+from spinn_front_end_common.utilities.constants import SIMULATION_N_BYTES
+
+
+# Identifiers for common regions
+CommonRegions = namedtuple(
+    "CommonRegions",
+    ["system", "provenance", "profile", "recording"])
+
+
+class PopulationMachineCommon(
```

Comment:
this cant be init'ed itself. thus its an abstract.

Reply:
I think if you extend an abstract you don't have to put the abstract here too (it might even break things)